torch/_functorch/aot_autograd.py

```diff
@@ -1812,7 +1794,7 @@ def add_dupe_args(args):
     def wrapped_flat_fn(*args):
         return flat_fn(*add_dupe_args(args))
 
-    compiled_fn = compiler_fn(wrapped_flat_fn, deduped_flat_args, aot_config)
+    compiled_fn = compiler_fn(wrapped_flat_fn, deduped_flat_args, aot_config, fw_metadata=fw_metadata)
```

Comment:
If you deduped the args, don't you have to update the `fw_metadata`?

Reply:
hmmm good catch.

I still want to avoid computing the metadata twice. There are probably a few ways of dealing with this, but here's what I'm thinking: `aot_dedupe()` will take in a metadata object that includes the analysis on dupe'd args, and only if necessary, return a new metadata object that is the same but with input metadata dedup'd.

Reply:
Yeah, having the deduper also dedupe the metadata soudns reasonable (if fiddly)

Reply:
this is done